src/storage/v3/request_helper.cpp

```diff
@@ -43,19 +44,74 @@ std::vector<Element> OrderByElements(Shard::Accessor &acc, DbAccessor &dba, Vert
     properties_order_by.reserve(order_bys.size());
 
     for (const auto &order_by : order_bys) {
-      const auto val =
+      auto val =
           ComputeExpression(dba, *it, std::nullopt, order_by.expression.expression, expr::identifier_node_symbol, "");
-      properties_order_by.push_back(val);
+      properties_order_by.push_back(std::move(val));
     }
     ordered.push_back({std::move(properties_order_by), *it});
   }
 
-  std::sort(ordered.begin(), ordered.end(), [compare_typed_values](const auto &pair1, const auto &pair2) {
+  std::sort(ordered.begin(), ordered.end(), [&compare_typed_values](const auto &pair1, const auto &pair2) {
     return compare_typed_values(pair1.properties_order_by, pair2.properties_order_by);
   });
   return ordered;
 }
 
+std::vector<GetPropElement> OrderByElements(DbAccessor &dba, std::vector<msgs::OrderBy> &order_by,
+                                            std::vector<GetPropElement> &&vertices) {
+  std::vector<Ordering> ordering;
```

Comment:
Once @42jeremy PR is merged, I will extract the common logic into functions and call them respectively.